Add documentation for CSR tables (#2295)
oneapi-src/oneDAL: docs/source/onedal/data-management/table/csr.rst

.. ******************************************************************************
.. * Copyright 2023 Intel Corporation
.. *
.. * Licensed under the Apache License, Version 2.0 (the "License");
.. * you may not use this file except in compliance with the License.
.. * You may obtain a copy of the License at
.. *
.. *     http://www.apache.org/licenses/LICENSE-2.0
.. *
.. * Unless required by applicable law or agreed to in writing, software
.. * distributed under the License is distributed on an "AS IS" BASIS,
.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
.. * See the License for the specific language governing permissions and
.. * limitations under the License.
.. *******************************************************************************/

.. _csr_table:

==================================

> Compressed sparse rows (CSR) table

Reviewer: ```suggestion
Compressed Sparse Rows (CSR) Table
```

==================================

Class ``csr_table`` is a subtype of a table type
for which the following is true:

> - The data within the table are sparse and stored in 3-array format.

Reviewer: ```suggestion
- The data within the table is sparse and stored in a 3-array format.
```

- All the columns have the same :capterm:`data type <Data type>`.

---------------------

> Programming interface

Reviewer: ```suggestion
Programming Interface
```

---------------------

> Refer to :ref:`API Reference: Compressed sparse rows table <api_csr_table>`.

Reviewer: ```suggestion
Refer to :ref:`API Reference: Compressed Sparse Rows Table <api_csr_table>` to learn more.
```
